workstreams/events_in_cicd/vocabulary.md

```diff
@@ -0,0 +1,108 @@
+These events are agnostic from any specific tool and are designed to fit a wide range of scenarios. 
+
+The phases covered by this proposal are:
+
+- **Source Code Version Control Events**: Events emitted by changes in source code or by the creation, modification or deletion of new repositories that hold source code.
+- **Continuous Integration Pipelines Events**: includes events related to building, testings, packaging and releasing software artifacts, usually binaries.
+- **Continuous Deployment Pipelines Events**: include events related with environments where the artifacts produced by the integration pipelines actually run. These are usually services running in a specific environment (dev, QA, production), or embeded software running in a specific hardware. 
+
+These phases can also be considered as different profiles of the vocabulary that can be adopted independently. 
+
+# Required Metadata for CD Events
+
+The following attributes are REQUIRED to be present in all the Events defined in this vocabulary:
+
+- **Event ID**: defines a unique identifier for the event
+- **Event Type**: defines a textual description of the event type, only event types described in this document are supported. All event types should be prefixed with `CD.`
+- **Event Source**: defines the context in which an event happened
+- **Event Timestamp**: defines the time when the event was produced
+
+The following sections list the events in different phases, allowing adopters to choose the events that they are more interested in.
+
+# Source Code Version Control Events
+
+This phase includes events related to changes in Source Code repositories that are relevant from a Continuous Delivery perspective.
+
+
+## Repository Events
+
+These events are related to Source Code repositories
+- **Repository Created Event**: a new Source Code Repository was created to host source code for a project
+- **Repository Modified Event**: a Source Code Repository modified some of its attributes, like location, or owner
+- **Repository Deleted Event**: a Source Code Repository was deleted and it is not longer available
+- **Branch Created Event**: a Branch inside the Repository was created 
+- **Branch Deleted Event**: a Branch inside the Repository was deleted
+- **Change Committed**: a change is commited to a branch
+- **Change Approved**: a change request is created by a user, that requires review and approval before being applied to an existing branch inside the repository
+- **Change Rejected**: a change was reviewed and rejected by a user
+- **Change Merged**: a change request has been merged to a branch in an existing repository
+
+
+Repository Events MUST include the following attributes:
+- **Event Type**: the type is restricted to include `CD.Repository**` prefix. For example `CD.Repository.Created` or `CD.Repository.ChangeApproved`
+- **Repository URL**: indicates the location of the source code repository
```

Comment:
Is this where we determine the type of the repository? E.g. "git:ssh://xyz"?

Reply:
@erkist let me add some more details there.